### PR TITLE
Align EC2 testdata models with `go.mod`

### PR DESCRIPTION
**Pull request by brycahta**
Issue #, if available: N/A, it is difficult to enhance code-generator+unit test, then compare results with `make build-controller` when models used by unit tests and the application differ

Description of changes:
* Updates EC2's testdata api models/shapes to align with the model version used by code-generator
* Updates EC2 unit tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/generate/code/resource_reference_test.go [hunk 1]
```diff
@@ -182,15 +182,15 @@ func Test_ReferenceFieldsPresent_NestedSliceOfStructsReference(t *testing.T) {
 	require.NotNil(crd)
 	expected :=
 		`if ko.Spec.Routes != nil {
-	for _, iter32 := range ko.Spec.Routes {
-		if iter32.GatewayRef != nil {
+	for _, iter35 := range ko.Spec.Routes {
+		if iter35.GatewayRef != nil {
```
Comment (RedbackThomson) [inline pkg/generate/code/resource_reference_test.go:186]: Oh interesting. Did this start failing just because the `api-2.json` was super out of date?

**Review by RedbackThomson — APPROVED** [review]

File: pkg/generate/code/resource_reference_test.go (hunk 1 above)
Comment (brycahta) [reply pkg/generate/code/resource_reference_test.go:186]: Yup, some fields were added to `CreateRoute`:

image

Comment (RedbackThomson) [reply pkg/generate/code/resource_reference_test.go:186]: Oh cool cool cool yep

**Review by A-Hilaly — APPROVED** [review]
/lgtm

**Pull request merged**